flink/src/main/java/org/apache/iceberg/flink/TableLoader.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.Serializable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopTables;
+
+/**
+ * Serializable loader to load an Iceberg {@link Table}.
+ * Flink needs to get {@link Table} objects in the cluster (for example, to get splits), not just on the client side.
+ * So we need an Iceberg table loader to get the {@link Table} object.
+ */
+public interface TableLoader extends Closeable, Serializable {
+
+  void open(Configuration configuration);
+
+  Table loadTable();
+
+  static TableLoader fromCatalog(CatalogLoader catalogLoader, TableIdentifier identifier) {
+    return new CatalogTableLoader(catalogLoader, identifier);
+  }
+
+  static TableLoader fromHadoopTable(String location) {
+    return new HadoopTableLoader(location);
+  }
+
+  class HadoopTableLoader implements TableLoader {
```

Comment:
Do you really need this? isn't this covered by the `CatalogTableLoader` using `hadoop`?

Reply:
I saw https://github.com/apache/iceberg/issues/1306 , I don't know if I misunderstood something.
It seems that a simple path can make it easier for users to use.

Reply:
Right, I think part of the discussion is that providing a table via this API does not guarantee the atomic commit on table changes, therefore we have `HiveCatalog` for tables that are backed in systems like `s3`. While `HadoopCatalog` that does not require HMS coordination since the filesystem itself has atomic renames.

I'm not sure which case specifically this specific Flink implementation will use, but as mentioned in https://github.com/apache/iceberg/issues/1306#issuecomment-671627947 it seems dangerous to provide a way to provide tables loaded with `HadoopTables` if there's no context of the underlying limitations.

Reply:
Yes, There are some comments in `HadoopCatalog`: `The HadoopCatalog requires that the underlying file system supports atomic rename.` So as `HadoopTables`.
But I think if users use `HadoopTables`, `HadoopCatalog` or `TableLoader.fromHadoopTable`, users should know, yeah, we need a file system supports atomic rename instead `S3` and etc...

So here is my thoughts:
- Just like Spark and Mapreduce. For a Flink `DataStream` programer, he/she can just specify a hadoop location path for reading and writing, or use iceberg `Catalog`.
- But for Flink SQL user, he/she must specify an iceberg `Catalog` to load tables.